Rubberduck.Core/UI/UnitTesting/Commands/AddTestModuleCommand.cs

```diff
@@ -154,98 +158,100 @@ protected override void OnExecute(object parameter)
         {
             var parameterIsModuleDeclaration = parameter is ProceduralModuleDeclaration || parameter is ClassModuleDeclaration;
 
-            var project = parameter as IVBProject ??
-                          (parameterIsModuleDeclaration ? ((Declaration) parameter).Project : GetProject());
-
-            if (project == null || project.IsWrappingNullReference)
+            using (var project = parameter as IVBProject ??
+                                 (parameterIsModuleDeclaration ? ((Declaration) parameter).Project : GetProject()))
```

Comment:
This using statement does not seem to be safe: if the `parameter` is an `IVBProject`, the lifetime has to be managed by the caller; if it is a module declaration, the project is coming from the declarations, which ultimately is managed by the projects repository. Only the very last option should be wrapped in a using statement.

Reply:
Good catch; fixed in my round 2 branch.